flutter/lib/device_info.dart

```diff
@@ -7,44 +7,80 @@ import 'package:mlperfbench_common/data/environment/environment_info.dart';
 import 'package:mlperfbench_common/data/environment/os_enum.dart';
 
 class DeviceInfo {
+  final String modelCode;
+  final String modelName;
+  final String manufacturer;
+
   static late final String nativeLibraryPath;
-  static late final String model;
-  static late final String manufacturer;
+  static late final DeviceInfo instance;
 
-  static EnvironmentInfo get environmentInfo {
-    return EnvironmentInfo(
-      osName: OsName.fromJson(Platform.operatingSystem),
-      osVersion: Platform.operatingSystemVersion,
-      manufacturer: DeviceInfo.manufacturer,
-      model: DeviceInfo.model,
-    );
-  }
+  DeviceInfo({
+    required this.manufacturer,
+    required this.modelCode,
+    required this.modelName,
+  });
 
   static Future<void> staticInit() async {
-    if (Platform.isAndroid) {
-      DeviceInfo.nativeLibraryPath = await _getNativeLibraryPath();
-    } else {
-      DeviceInfo.nativeLibraryPath = '';
-    }
+    DeviceInfo.nativeLibraryPath = await _makeNativeLibraryPath();
+    DeviceInfo.instance = await createFromEnvironment();
+  }
 
+  static Future<DeviceInfo> createFromEnvironment() async {
     if (Platform.isIOS) {
-      final deviceInfo = DeviceInfoPlugin();
-      final iosInfo = await deviceInfo.iosInfo;
-
-      DeviceInfo.manufacturer = 'Apple';
-      DeviceInfo.model = iosInfo.utsname.machine;
+      return _makeIosInfo();
     } else if (Platform.isWindows) {
-      DeviceInfo.manufacturer = 'Microsoft';
-      DeviceInfo.model = 'Unknown PC';
+      return _makeWindowsInfo();
     } else if (Platform.isAndroid) {
-      final deviceInfo = await DeviceInfoPlugin().androidInfo;
-
-      DeviceInfo.manufacturer = deviceInfo.manufacturer;
-      DeviceInfo.model = deviceInfo.model;
+      return _makeAndroidInfo();
     } else {
       throw 'Could not define platform';
     }
   }
+
+  static Future<String> _makeNativeLibraryPath() async {
+    if (Platform.isAndroid) {
+      return await _getNativeLibraryPath();
+    } else {
+      return '';
+    }
+  }
+
+  static Future<DeviceInfo> _makeIosInfo() async {
+    final deviceInfo = DeviceInfoPlugin();
+    final iosInfo = await deviceInfo.iosInfo;
+
+    return DeviceInfo(
+      manufacturer: 'Apple',
+      modelCode: iosInfo.utsname.machine,
+      modelName: '',
+    );
+  }
+
+  static Future<DeviceInfo> _makeAndroidInfo() async {
+    final deviceInfo = await DeviceInfoPlugin().androidInfo;
+    return DeviceInfo(
+      manufacturer: deviceInfo.manufacturer,
+      modelCode: deviceInfo.model,
+      modelName: '',
+    );
+  }
+
+  static Future<DeviceInfo> _makeWindowsInfo() async {
+    return DeviceInfo(
+      manufacturer: 'Microsoft',
```

Comment:
Is `Microsoft` the manufacturer for all Windows devices?

Reply:
No.
I also caught this issue when working on #512 but forgot to push it into this PR.